### PR TITLE
Move to using Build Props file and central package management.

### PR DESCRIPTION
**Pull request by mhbuck**
## Changes

 - This moves the project over to using a build props file
 - Standardise the language version across all projects and minor test clean up
 - Move to central package management.

## Checklist

For significant changes, please ensure that the following have been completed (delete if not relevant):

* [ ] I have [reviewed my own pull request](https://getakka.net/community/contributing/index.html#review-your-own-pull-requests).

I am unsure if the build system needs some things to be upgraded as well or not. Similar to 
- https://github.com/akkadotnet/Akka.Streams.Kafka/pull/337
- https://github.com/akkadotnet/Akka.Persistence.PostgreSql/pull/179

### CODE REVIEW

File: src/Akka.Persistence.MongoDb.Tests/Hosting/MongoDbJournalOptionsSpec.cs [hunk 1]
```diff
@@ -131,7 +131,7 @@ public void JournalOptionsTest()
         [Fact(DisplayName = "MongoDbJournalOptions should be bindable to IConfiguration")]
         public void JournalOptionsIConfigurationBindingTest()
         {
-            var stream = new MemoryStream(Encoding.UTF8.GetBytes(Json));
+            using var stream = new MemoryStream(Encoding.UTF8.GetBytes(Json));
```
Comment (mhbuck) [inline src/Akka.Persistence.MongoDb.Tests/Hosting/MongoDbJournalOptionsSpec.cs:134]: Now all projects use the same langversion can clean up old items from the tests

File: src/Directory.Build.props [hunk 2]
```diff
@@ -12,22 +12,19 @@
     <GenerateDocumentationFile>true</GenerateDocumentationFile>
     <Description>Akka Persistence journal and snapshot store backed by MongoDB database.</Description>
     <NoWarn>$(NoWarn);CS1591</NoWarn>
+    <LangVersion>latest</LangVersion>
```
Comment (mhbuck) [inline src/Directory.Build.props:15]: Most projects seem to be using latest. Can change this to what ever is appropriate.

File: src/Directory.Build.props [hunk 3]
```diff
@@ -12,22 +12,19 @@
...
   </PropertyGroup>
   <PropertyGroup>
     <NetCoreTestVersion>netcoreapp3.1</NetCoreTestVersion>
     <NetTestVersion>net6.0</NetTestVersion>
     <NetFrameworkTestVersion>net471</NetFrameworkTestVersion>
     <NetStandardLibVersion>netstandard2.0</NetStandardLibVersion>
 
+    <AkkaVersion>1.5.8</AkkaVersion>
```
Comment (mhbuck) [inline src/Directory.Build.props:23]: I have seen some projects have two variables for Akka and Akka Hosting. Not sure if required
https://github.com/akkadotnet/Akka.Persistence.Sql/blob/8052a17c75a678ca8052923a3d8ee43a19fb2bd7/src/Directory.Build.props#L24C1-L24C53

File: src/Directory.Build.props [hunk 4]
```diff
@@ -12,21 +12,22 @@
...
     <XunitVersion>2.4.2</XunitVersion>
     <TestSdkVersion>17.6.3</TestSdkVersion>
-    <AkkaVersion>1.5.8</AkkaVersion>
-	<FluentAssertionsVersion>6.11.0</FluentAssertionsVersion>
+    <FluentAssertionsVersion>6.11.0</FluentAssertionsVersion>
   </PropertyGroup>
   <!-- SourceLink support for all Akka.NET projects -->
   <ItemGroup>
-    <PackageReference Include="Microsoft.SourceLink.GitHub" Version="1.1.1" PrivateAssets="All" />
+    <PackageReference Include="Microsoft.SourceLink.GitHub" PrivateAssets="All" />
```
Comment (mhbuck) [inline src/Directory.Build.props:30]: Note I have seen some other repos that have moved to central package management have the `Microsoft.SourceLink.GitHub` in the `Directory.Packages.props` file but no reference either in the csproj or `Directory.Build.props`

I have noticed it in `Akka.Persistence.Sql` and `Akka.Persistence.PostgreSql` repos

**Review by Aaronontheweb — APPROVED** [review]
LGTM

**Pull request merged**